pkg/test/test_data/teststep-apply/00-assert.yaml

```diff
@@ -0,0 +1,14 @@
+# Verify that the pods were created.
+apiVersion: v1
+kind: Pod
+metadata:
+  name: hello
+status:
+  phase: Pending
```

Comment:
why verify `Pending` phase? why not just remove `status`? 
my question is this is because of current test setup that it stays in `Pending` or that's the desired assertion?

Reply:
it is the desired assertion for tests in the mocked control plane (which is also consistent with the other tests in this test suite.

Reply:
Pods will never come out of pending in the local control plane mode since there's no kube-controller-manager running.

Reply:
but if we decide to run those tests in a kubernetes cluster we will need to change tests?
my question was if asserting `Pending` status we are verifying a mocked control-plane behavior? when the intent I see from test assertion is to verify that object is created

Reply:
tests do depend on knowing what env it will be run against... there are conditions you will never get to in a mock control plane... the expectation should be that tests against a mock control plane should be for an initial state... and tests against a cluster are end states (or controlled states).  It is too easy to "miss" a state.  beginning or intermediate states could be completely missed in a cluster resulting in flaky behavior.